lib/rules/color-function-notation/index.mjs

```diff
@@ -67,9 +67,13 @@ const rule = (primary, secondaryOptions) => {
 
 				if (!LEGACY_NOTATION_FUNCTION_NAME.test(value)) return;
 
-				if (primary === 'modern' && !hasCommas(node)) return;
+				const hasEnoughArguments = nodes.length >= 5;
 
-				if (primary === 'legacy' && hasCommas(node)) return;
+				if (!hasEnoughArguments) return;
+
+				if (primary === 'modern' && !isLikelyLegacy(nodes)) return;
```

Comment:
Maybe assign the result of `isLikelyLegacy(nodes)` to a variable so that it can be reused?

Reply:
nvm :)

Only one is ever executed because `primary` is always `modern` or `legacy`.